src/compas_timber/connections/french_ridge_lap.py

```diff
@@ -40,6 +40,7 @@ def __init__(self, beam_a=None, beam_b=None, **kwargs):
         super(FrenchRidgeLapJoint, self).__init__(beams=(beam_a, beam_b), **kwargs)
         self.beam_a = beam_a
         self.beam_b = beam_b
+        self.elements.extend([beam_a, beam_b])
```

Comment:
Joints can include fasteners and plates (and other elements), not just beams. 

I think we should generalize and systematize the elements in the joint so that we can consolidate the (de)serialization, instead of having every joint do it's own beam serialization only because we have a different name for the beams. 

I would like to have:
in parent class:
`Joint.elements = []`

in child class:
```
 JointType.__init__(beam_a, beam_b, **kwargs):
    self.elements.extend([beam_a, beam_b)

@property           #we can have a setter, too
def beam_a(self):
    return self.elements[0]

@property
def beam_b(self):
    return self.elements[1]
```

Reply:
if I get it right this is about inverting the logic of how we reference elements in a Joint.
this would be really cool if it allowed us to eliminate the `__data__` in each joint implementation but since joints have other attributes that need be serialized, we cannot yet get rid of this. which makes this change for me rather an arbitrary decision.

maybe we can debate which one's better but that's not a prerequisite for multi element joints, plus, you're suggesting here a rework of `Joint` which might affect other PRs, and it doesn't seem complete in this PR.